packages/neuron-ui/src/components/BroadcastTransaction/index.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { useCallback, useMemo, useState } from 'react'
+import { useTranslation } from 'react-i18next'
+import { useNavigate } from 'react-router-dom'
+import { isSuccessResponse, RoutePath, isMainnet as isMainnetUtil, useGoBack, getExplorerUrl } from 'utils'
+import Dialog from 'widgets/Dialog'
+import AlertDialog from 'widgets/AlertDialog'
+import { useDispatch, useState as useGlobalState } from 'states'
+import { broadcastTransactionOnly, OfflineSignStatus, openExternal, getTransactionList } from 'services/remote'
+import { ReactComponent as HardWalletIcon } from 'widgets/Icons/HardWallet.svg'
+
+import styles from './broadcastTransaction.module.scss'
+
+const BroadcastTransaction = () => {
+  const {
+    app: { loadedTransaction = {} },
+    chain: { networkID },
+    settings: { networks },
+    wallet,
+  } = useGlobalState()
+
+  const [isBroadcasting, setIsBroadcasting] = useState(false)
+  const [broadcastedTxHash, setBroadCastedTxHash] = useState<string | null>('')
+  const [t] = useTranslation()
+  const dispatch = useDispatch()
+  const [errMsg, setErrMsg] = useState('')
+  const isMainnet = isMainnetUtil(networks, networkID)
+
+  const { filePath, json } = loadedTransaction
+
+  const jsonContent = useMemo(() => {
+    return JSON.stringify(json, null, 2)
+  }, [json])
+
+  const signStatus: OfflineSignStatus = json.status
+
+  const isSigned = useMemo(() => signStatus === OfflineSignStatus.Signed, [signStatus])
+
+  const onBack = useGoBack()
+
+  const navigate = useNavigate()
+  const onBroadcast = useCallback(async () => {
+    if (broadcastedTxHash) {
+      openExternal(`${getExplorerUrl(isMainnet)}/transaction/${broadcastedTxHash}`)
+      return
+    }
+
+    setIsBroadcasting(true)
+
+    const res = await broadcastTransactionOnly({
+      ...json,
+    })
+
+    setIsBroadcasting(false)
+
+    if (isSuccessResponse(res)) {
+      if (!wallet?.id) {
+        setBroadCastedTxHash(res.result)
+        return
+      }
+
+      if (res.result) {
+        getTransactionList({
+          walletID: wallet.id,
+          pageNo: 1,
+          pageSize: 1,
```

Comment:
Maybe the size could be a bit larger for fault tolerance. Or the `search` function in the history could be used to check if the `res.result` is in the history.

Reply:
> Maybe the size could be a bit larger for fault tolerance. Or the `search` function in the history could be used to check if the `res.result` is in the history.

The logic of methods related to search in history is not suitable for the current scenario.

https://github.com/nervosnetwork/neuron/blob/4907867165926ada3240b390e27c808c1103324b/packages/neuron-ui/src/components/History/index.tsx#L48C21-L48C37

Reply:
How about using `get transaction by wallet id and tx hash` at https://github.com/nervosnetwork/neuron/blob/develop/packages/neuron-wallet/src/controllers/api.ts#L465

Reply:
https://github.com/nervosnetwork/neuron/blob/4907867165926ada3240b390e27c808c1103324b/packages/neuron-wallet/src/controllers/transactions.ts#L97C13-L97C13

As long as the wallet and transaction exist, transaction information will be returned regardless of whether the transaction is related to that wallet or not.

Reply:
My fault, but `get-transaction-list` should work, it fetches transactions belong to a wallet by a specific keyword, and the keyword could be the tx hash

Reply:
> My fault, but `get-transaction-list` should work, it fetches transactions belong to a wallet by a specific keyword, and the keyword could be the tx hash

Yes, it is now using `get-transaction-list` and hash for queries.

Reply:
LOL, I must have to buy a new glasses.